src/server.py

```diff
@@ -238,12 +234,25 @@ def _build_pipeline(self):
 
     def __bus_message_cb(self, unused_bus, message):
         if message.type == Gst.MessageType.EOS:
+            self.pipeline.set_state(Gst.State.NULL)
+            self.emit("eos")
+        elif message.type == Gst.MessageType.SEGMENT_DONE:
+            if message.src != self.pipeline:
+                return
             if self.loop and not self._stop_loop:
-                self._remove_fade_in()
-                self.seek(0.0)
+                self.seek(0.0, flags=Gst.SeekFlags.SEGMENT)
             else:
-                self.pipeline.set_state(Gst.State.NULL)
-                self.emit("eos")
+                # Cancel the SEGMENT seek.
+                self.seek()
+                self.pipeline.send_event(Gst.Event.new_eos())
+        elif message.type == Gst.MessageType.ASYNC_DONE:
+            if message.src != self.pipeline:
+                return
+            if self.loop and not self._is_initial_seek:
```

Comment:
Can you do the segment seek already on the first loop so you don't have to keep track of whether it is the initial seek or not?

Reply:
There are two points in which I find possible to do this segment SEEK: on ASYNC_DONE or while it is playing (we would have to decide on a point... which point? we don't want to take this decision) sending a SEGMENT seek with no position (seek_type=NONE). 
The reason why I send a SEGMENT seek there is to avoid EOS being emitted at the end of the first loop, and so avoid the clock getting restarted.

Reply:
That is what I am doing.

Reply:
It looks like you are doing it when the first ASYNC_DONE message is emitted. Does that somehow happen at time 0?